lib/glific/contacts/import_worker.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Glific.Contacts.ImportWorker do
+  @moduledoc """
+  Worker for processing contact chunks.
+  """
+  require Logger
+
+  use Oban.Worker,
+    queue: :default,
+    max_attempts: 2,
+    priority: 2
+
+  alias Glific.{
+    Contacts.Import,
+    Jobs.UserJob,
+    Repo
+  }
+
+  @doc """
+  Creating new job for each chunk of contacts.
+  """
+  @spec make_job(list(), map(), non_neg_integer(), non_neg_integer()) :: :ok
+  def make_job(chunk, params, user_job_id, index) do
+    __MODULE__.new(%{contacts: chunk, params: params, user_job_id: user_job_id})
+    |> Oban.insert(schedule_in: index)
+  end
+
+  @doc """
+  Standard perform method to use Oban worker.
+  """
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"contacts" => contacts, "params" => params, "user_job_id" => user_job_id}}) do
+    params = %{
+      organization_id: params["organization_id"],
+      user: %{
+        roles: Enum.map(params["user"]["roles"], &String.to_existing_atom/1),
+        upload_contacts: params["user"]["upload_contacts"]
+      }}
+
+    Repo.put_process_state(params.organization_id)
+
+    errors = Enum.reduce(contacts, %{}, fn contact, acc ->
+      phone_number = Map.get(contact, "phone")
+      acc = validate_phone(acc, phone_number)
+      acc
+    end)
+
+    user_job = Repo.get(UserJob, user_job_id)
+    tasks_done = user_job.tasks_done + 1
+    updated_errors = Map.merge(user_job.errors || %{}, errors)
+    UserJob.update_user_job(user_job, %{tasks_done: tasks_done, errors: updated_errors})
+
+    contacts = Enum.map(contacts, fn contact ->
+      for {key, value} <- contact, into: %{}, do: {String.to_existing_atom(key), value}
+    end)
```

Comment:
We can avoid this, we are doing this to convert string keys to atom. Instead we can adjust the actual `process_data` function in import.ex to accept String keys, since that function is anyway used only by worker